src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -110,14 +111,31 @@ public <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToCQL(
       } else if (columnType instanceof SetType st) {
         List<JSONCodec<?, ?>> valueCodecCandidates = codecsByCQLType.get(st.getElementType());
         if (valueCodecCandidates != null) {
-          // Almost there! But go avoid ClassCastException if input not a JSON Array need this check
+          // Almost there! But to avoid ClassCastException if input not a JSON Array need this check
           if (!(value instanceof Collection<?>)) {
             throw new ToCQLCodecException(value, columnType, "no codec matching value type");
           }
           return (JSONCodec<JavaT, CqlT>)
               CollectionCodecs.buildToCQLSetCodec(valueCodecCandidates, st.getElementType());
         }
         // fall through
+      } else if (columnType instanceof MapType mt) {
+        List<JSONCodec<?, ?>> valueCodecCandidates = codecsByCQLType.get(mt.getValueType());
+        if (valueCodecCandidates != null) {
+          // Must check key type: only text/ascii supported
+          if (!isSupportedMapKeyType(mt.getKeyType())) {
```

Comment:
we have this check for supported key types in a few places. Do you also have a check for supported value types ? 
Do you have the same for the valueTypes for lists , sets, and vectors ? 

if so can you put them public in places that make sense and I can call them from higher up when working out if we support the schema for a table 

e.g. I have this in a PR for working out if we support the map schema for a table 

```java
    public static boolean isCqlTypeSupported(MapType cqlMapType) {
      Objects.requireNonNull(cqlMapType, "cqlMapType must not be null");

      // cannot be frozen
      if (cqlMapType.isFrozen()) {
        return false;
      }
      // keys must be text or ascii, because keys in JSON are string
      if (!(cqlMapType.getKeyType() == DataTypes.TEXT
          || cqlMapType.getKeyType() == DataTypes.ASCII)) {
        return false;
      }
      // must be a primitive type value
      return cqlMapType.getValueType() instanceof PrimitiveType;
    }

```

Reply:
Ok: I'll do a follow-up ticket for this, need to think it through. One concern I have with "isXxx()" methods like this is that they need to find most of the info, then throw it away -- so, f.ex when checking if Map type supported, we need to lookup value codec (since that is how we know if type is supported), throw it away. And then in case of constructing Map codec we need to again lookup value codec.

Reply:
Oh. One bigger concern: supported how? To read, write, create column?